osu.Game/Beatmaps/BeatmapUpdater.cs

```diff
@@ -44,7 +52,13 @@ public void Queue(Live<BeatmapSetInfo> beatmap)
         /// <summary>
         /// Run all processing on a beatmap immediately.
         /// </summary>
-        public void Process(BeatmapSetInfo beatmapSet) => beatmapSet.Realm.Write(r => Process(beatmapSet, r));
+        public void Process(BeatmapSetInfo beatmapSet)
+        {
+            if (beatmapSet.Realm.IsInTransaction)
+                Process(beatmapSet, beatmapSet.Realm);
+            else
+                beatmapSet.Realm.Write(r => Process(beatmapSet, r));
```

Comment:
> doesn't seem like anywhere with current code

That sounds wrong.. did you try making the change and running tests? At very least, editor save should fail.

> why is this not an overload of the Write() method or done inside the Write() method itself?

I could make `Write` just-work in these scenarios, but I'm not 100% sure that's what we want. Maybe we can go with it and see if it bites us later on?

Reply:
I see, so the change that "broke" this is the new `ProcessBeatmap` callback doesn't have a `Realm` object so the `Process(set, realm)` overload can't be used.

I dunno, perhaps I'm being too picky, but this is the first time this sort of interaction is being done and it looks like it's been thrown in here because some caller required this particular design, rather than the system being well designed in the first place.

At the very least this requires a comment. Such as `when beatmaps are saved, they're reprocessed from inside an existing transaction, and nested transactions aren't supported by Write()`.

Reply:
Yeah, that's the one.

I'll go through this again, I agree it's not great being here and sets a bad precedent. @bdach also mentioned wanting `Write` to allow nested transactions, so maybe that's the best way forward (until we run into an issue with it)?

Reply:
I'm not ready to make calls on this. If there's a reason it hasn't been done yet, there must be some concern you've had about it which I don't harbour. I'm fine with just a comment, but just pointing out that this code looks dodgy.

Reply:
My main concern is that once you remove the explicit signalling of being in a transaction, it becomes difficulty to tell if you are or not. Which means we may start calling (or thinking we need to call) `BeginTransaction` or `realm.Write` unnecessary, or as a precaution.

Reply:
I think specifically with these extension methods it's probably fine to make the change for now, so I'm going to do that. Can always reassess in the future.